V5 style refinements

domyen opened the pull request:
Note: Based on #5426, #5425, #5428 (merge this one first)
Issue: More progress on decomposing stories and components in the UI

## What I did
- Added placeholder stories for Sidebar
- Added Global styles on `body`
- Refactored Sidebar explorer, treeview, and StoriesExplorer into one file
- Removed unused files leftover from Tom's refactor
- Minor tweaks to styles and how we organize Storybook's own stories

Commit:
Add stories for Sidebar

Commit:
Reorganize stories to mirror directory structure

Commit:
Remove unnecessary Badge stories

Commit:
Update icon, StorybookIcon, and StorybookLogo components

Commit:
Tweak sidebar wording to be less redundant "components" -> "basics"

Commit:
Export Badge with @storybook/components

Commit:
Changed settings pages header icon and cleaned up some legacy icon exports

Commit:
Merge branch 'next' into v5-style-refinements

Commit:
Merge remote-tracking branch 'origin/add-sidebar-stories' into v5-style-refinements

Commit:
Merge remote-tracking branch 'origin/add-preview-stories' into v5-style-refinements

Commit:
Merge remote-tracking branch 'origin/tech/about-screen-fixes' into v5-style-refinements

Commit:
Remove duplicate file and story

Commit:
Refine SidebarItem loading state

Commit:
Added closeAlt icon and used in on settings screens

Commit:
Minor styling tweak for SidebarSubheading

Commit:
Update snapshots and fix tests

Commit:
ADD Global styles to app and storybook

Commit:
REFACTOR Explorer, StoriesExplorer & Treeview for ergonomics

Commit:
Remove unused file

Commit:
Remove missing component in enzyme test

Commit:
Change SidebarSubheading to div instead of span

Commit:
Merge branch 'next' into v5-style-refinements

tmeasday commented on lib/ui/src/components/sidebar/SidebarList.js:
```diff
@@ -0,0 +1,8 @@
+
+const SidebarList = props => <Heading {...props} />;
+
+export default SidebarList;
```
What is this component? I think what you call the `SidebarList` is the `StoriesExplorer` which already is built and has these stories

tmeasday commented on lib/ui/src/components/sidebar/SidebarSearch.stories.js:
```diff
@@ -1,18 +1,8 @@
-);
+storiesOf('UI|Sidebar/SidebarSearch', module).add('default', () => (
+  <SidebarSearch>TODO SidebarSearch</SidebarSearch>
+));
```
OI! What did you do to my stories!!

tmeasday reviewed (changes requested):
Seems good apart from (a) surplus component and (b) crippling of the search?

Commit:
Remove body styles from theme type

cc @ndelangen @domyen -- I don't really have the full context here, just trying to get this to build.

I don't know if this is the right move (maybe we want that?) but doing this to keep things moving

Commit:
REVERT bad merge to `SidebarSearch`

And remove redundant `SidebarList`

Commit:
This file should be deleted

Commit:
FIX broken stories.

I'm not sure these stories/component are actually used?

Commit:
UPDATE snapshots

Commit:
UPDATE image snapshots

Commit:
Merge remote-tracking branch 'origin/next' into v5-style-refinements

Commit:
Merge remote-tracking branch 'origin/next' into v5-style-refinements

ndelangen committed:
FIX package.json & yarn.lock

ndelangen commented:
https://circleci.com/workflow-run/5d2d070c-1e99-4d0d-a191-d5c5c6738d98

ndelangen commented:
@domyen I think there's a few regressions here, also the fonts look a bit too big, (maybe that's on purpose?)

Do you want to fix them in this branch?

domyen commented:
I'm working on the styling infra so expect a lot of visual regressions while work is in progress. 

Though I'd prefer to merge into `next` often, it might not make sense given the breadth of visual changes and all the story renaming/editing/removing. How about I keep this branch up to date with `next` and submit the PR when closer to stable? @tmeasday 

@ndelangen Yea I'd prefer to fix them on this branch.

Commit:
Fix SidebarItem warning for loading state

Commit:
Fix dupe treeview.mockdata reference

Commit:
Tweak styling for SidebarHeading & allow it to be styled by emotion via props

Commit:
Improve SidebarItem styling & add more stories

This removes some of the layout theming variables in favor of styling clarity. We can add them back if we decide to allow themers to adjust layout primitives.

Commit:
Minor styling tweaks to Sidebar subcomponents

Commit:
Refactor styling for Sidebar & change name of StoriesExplorer component

Change to a more specific name since StoriesExplorer is only used in one place at the moment. Willing to change if/when needed.

Commit:
Style SidebarSearch & add input reset UX

Commit:
Rename a SidebarItem story

Commit:
SidebarSearch fix border and improve story

Commit:
Add Sidebar loading state and refine SidebarHeading more

Commit:
Replace Button component with the styled component from /frontpage

Commit:
Minor styling and design tweaks

Commit:
Add `tooltips` component and subcomponents from storybooks/frontpage

Based on the same tech as `menu`

Commit:
Sidestep known @storybook/theming emotion bug https://github.com/emotion-js/emotion/issues/1203

Commit:
Add ListItemIcon which provides an icon/img/placeholder for ListItem

Commit:
Forgot to add recompose as a dep in @storybook/components

Commit:
Replace Menu with Tooltip

Pass data instead of composing the menu in situ each time its used

Commit:
Minor tweaks

Commit:
Merge branch 'next' into v5-style-refinements

Commit:
Update placeholder styles and API

Commit:
Capitalize proper noun

Commit:
NotificationItem and NotificationList,  simplify API and style

ndelangen committed:
Merge branch 'next' into v5-style-refinements

Commit:
Rename panel_actionbar to ActionBar & update API (see comment)

API updated so consumers don't have to compose ActionBar + ActionButton in situ every usage. This removes needing to export 2 components from @storybook/components and then import them in addons and elsewhere in the UI. What do you think @ndelangen @tmeasday?

Commit:
Make 'hoverable' animation part of the theme for use in interactive elements like Button, <a>, etc.

Commit:
Minor non-breaking tweaks

ndelangen committed:
Merge branch 'next' into v5-style-refinements

# Conflicts:
#	addons/backgrounds/src/Tool.js
#	addons/viewport/src/Tool.js

ndelangen committed:
FIX snapshots

ndelangen committed:
Merge branch 'v5-style-refinements' of github.com:storybooks/storybook into v5-style-refinements

ndelangen committed:
FIX snapshots

ndelangen committed:
FIX Tools after merge

ndelangen committed:
FIX initial id of preview (again)

ndelangen committed:
FIX button svg

ndelangen committed:
FIX animation on layout

Commit:
Add DocumentFormatting component

Commit:
Refactor the Placeholder API for better backwards compat

Commit:
Minor tweaks and changes

ndelangen committed:
REFACTOR Tooltip so it's allow for colorization

Commit:
Remove unused/underused `typography` components  & Add link styles & update use cases

Commit:
Add story that calls out font-size primitives (design tokens)

Commit:
Remove opacity setting that made code hard to read when using `syntaxhighlighter` component

Commit:
Styling bugs

Commit:
Refactor SettingsFooter out of About and Shortcut pages

Commit:
Merge branch 'v5-style-refinements' of https://github.com/storybooks/storybook into v5-style-refinements

Commit:
Use the new story format in SettingsFooter

ndelangen committed:
FIX tooltip things

ndelangen committed:
Merge branch 'v5-style-refinements' of github.com:storybooks/storybook into v5-style-refinements

ndelangen committed:
FIX warnings from emotion about '":nth-child" is potentially unsafe'

ndelangen committed:
FIX portal issue

ndelangen committed:
FIX double tooltip content

ndelangen committed:
Merge branch 'next' into v5-style-refinements

Commit:
Minor tweaks to styling all over the place

Commit:
Consolidate settings components and shortcuts page, remove extraneous DOM elements

Commit:
Merge branch 'v5-style-refinements' of https://github.com/storybooks/storybook into v5-style-refinements

Commit:
settings/components no longer exist, removes associated shallow tests

Commit:
Rename `grid` to `Spaced` because that's what the export is called

Commit:
Fix console error with SidebarItem loading state (no `name` prop)

Commit:
Update snapshots

Commit:
Rename Spaced story name

Commit:
Add syntaxhighlighter `padded` story

Commit:
Give tabs a default font-size

Commit:
Fix regression: Use the buttons from form instead of the other storybook buttons

Temporarily fixes regression from having 2 types of buttons, need to consolidate in the future

Commit:
Use Link component instead of raw <a>

Commit:
Fix tests and warnings

ndelangen committed:
FIX build

ndelangen committed:
Merge branch 'v5-style-refinements' of github.com:storybooks/storybook into v5-style-refinements

ndelangen committed:
Merge branch 'next' into v5-style-refinements

# Conflicts:
#	lib/ui/package.json

Commit:
Refactor DocumentFormatting to better mirror GitHub default Markdown

Commit:
Tweak Tabs to have a non-specific empty message

Commit:
Move ListItemIcon into /sidebar

Commit:
Use a styled Link when possible

Commit:
Update snapshot

Commit:
Merge branch 'v5-style-refinements' of https://github.com/storybooks/storybook into v5-style-refinements

ndelangen committed:
FIX snapshots

ndelangen committed:
FIX snapshots

ndelangen committed:
FIX typo

ndelangen committed:
IMPROVE the story for spaced, so chromatic captures it all

ndelangen committed:
Merge branch 'next' into v5-style-refinements

ndelangen committed:
REFACTOR global styles into reset & global && FIX background tool && ADD preview background to theme

ndelangen committed:
FIX snapshots

ndelangen committed:
ADD isLeaf to storiesHash && use isLead to render different Link in treeview

This fixes the navigating to non-existent stories issue.

ndelangen committed:
FIX mockdata for sidebar

ndelangen committed:
FIX menu not expanding

Commit:
Theme Storysource && update SyntaxHighlighter to accept Emotion styling

Commit:
Update styles in a11y addon

Commit:
Minor styling and copy edits

Commit:
Merge branch 'v5-style-refinements' of https://github.com/storybooks/storybook into v5-style-refinements

Commit:
Audit the theme, remove unused, update used (WIP)

Commit:
Update: DocumentFormatting static colors to use theme colors

Commit:
Update token for theme.mainTextSize

Commit:
Update global style variables

Commit:
Make Form.Button customize Button instead of adding a whole new button style

Commit:
Update styling for Knobs.Color + Form.Field + stories

Commit:
Use theme-level borderRadius instead of hardcoding

Commit:
Update Shortcuts screen with Form components

Commit:
Change global typography sizes/weights to numbers from strings

Commit:
Minor styling tweaks

Commit:
Remove unused Form stories (we don't want people to use them anyways!)

Commit:
Add a nicer empty message for notes that links people to the notes docs

Commit:
Fix StorybookIcon svg misrendering

Commit:
Move single-use Preview component into the /UI dir

Commit:
Delete unused HighlightButton

Commit:
Tweak layout of ListItemIcon slightly per the new Tooltip

Commit:
Update snapshot tests

Commit:
Fix icon rendering in Chrome

Commit:
Add title tags to IconButtons in the toolbars

Commit:
Polish

Commit:
Update snapshots

ndelangen committed:
Merge branch 'next' into v5-style-refinements

ndelangen committed:
UPGRADE react sibling packages alongside react

ndelangen committed:
WIP

Commit:
Change theme var names

Commit:
Rework brand to expect a DOM element (i.e., an SVG)

Commit:
Update snapshot test

Commit:
Remove unneccessary font declarations

Commit:
Remove unused Table component

Commit:
Style Actions addon better && re-add syntax highlighting!

Commit:
Remove some Button :focus styles

Commit:
Minor cleanup

shilman committed:
Dark theme WIP (get build working)

shilman commented:
@domyen merging this now to get a beta out
